src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEntityTypeDefinition.cs

```diff
@@ -139,5 +161,30 @@ private IEnumerable<IEdmStructuralProperty> ComputeDeclaredKey()
 
             return null;
         }
+
+        private IEnumerable<KeyValuePair<IEdmStructuralProperty, string>> ComputeDeclaredKeyAlias()
+        {
+            if (this.entity.Key != null)
+            {
+                List<KeyValuePair<IEdmStructuralProperty, string>> keyAlias = new List<KeyValuePair<IEdmStructuralProperty, string>>();
+                foreach (CsdlPropertyReference keyProperty in this.entity.Key.Properties)
+                {
+                    string propertyName = "";
+                    if (!String.IsNullOrEmpty(keyProperty.Alias))
+                    {
+                        string[] keyProperties = keyProperty.PropertyName.Split('/');
+                        propertyName = keyProperties[1];
+
+                        IEdmStructuralProperty structuralProperty = this.FindProperty(propertyName) as IEdmStructuralProperty;

```

Comment:
If `propertyName` is the part of a reference to a property of a complex property or related entity, then I assume that means that the reference property is not declared on this entity. If that's the case, how would `this.FindProperty()` find that property?

Furthermore, if you strip out the slashes, then it's possible you might end up with a different property altogether.

For example, let's say we have the following entity:
```xml
<EntitType Name="Foo">
  <Key>
     <PropertyRef Alias="ID" Name="Complex/Bar">
  </Key>
  <Property Name="Bar" Type="Edm.Int32" Nullable="False" />
  <Property Name="Complex" Type="NS.Complex" Nullable="False" />
</EntityType>
<ComplexType Name="Complex">
   <Property Name="Bar" Type="Edm.Int32" Nullable="False" />
</ComplexType>
```
If the property name is "Complex/Bar" then you know that it uniquely refers to the `Bar` property of the `Complex` property of the entity. But if you split the property name and just remain with `Bar` then you might end up with the wrong `Bar` property cause now it's ambiguous. It's also possible that the entity has multiple complex properties with a property called `Bar`.